specs/core/0_beacon-chain.md

```diff
@@ -1448,21 +1449,25 @@ def exit_validator(state: BeaconState, index: ValidatorIndex) -> None:
 #### `slash_validator`
 
 ```python
-def slash_validator(state: BeaconState, index: ValidatorIndex) -> None:
+def slash_validator(state: BeaconState, slashed_index: ValidatorIndex, whistleblower_index: ValidatorIndex=None) -> None:
     """
-    Slash the validator with index ``index``.
+    Slash the validator with index ``slashed_index``.
     Note that this function mutates ``state``.
     """
-    validator = state.validator_registry[index]
-    exit_validator(state, index)
-    state.latest_slashed_balances[get_current_epoch(state) % LATEST_SLASHED_EXIT_LENGTH] += get_effective_balance(state, index)
-
-    whistleblower_index = get_beacon_proposer_index(state, state.slot)
-    whistleblower_reward = get_effective_balance(state, index) // WHISTLEBLOWER_REWARD_QUOTIENT
-    increase_balance(state, whistleblower_index, whistleblower_reward)
-    decrease_balance(state, index, whistleblower_reward)
-    validator.slashed = True
-    validator.withdrawable_epoch = get_current_epoch(state) + LATEST_SLASHED_EXIT_LENGTH 
+    exit_validator(state, slashed_index)
+    state.validator_registry[slashed_index].slashed = True
+    state.validator_registry[slashed_index].withdrawable_epoch = get_current_epoch(state) + LATEST_SLASHED_EXIT_LENGTH
+    slashed_balance = get_effective_balance(state, slashed_index)
+    state.latest_slashed_balances[get_current_epoch(state) % LATEST_SLASHED_EXIT_LENGTH] += slashed_balance
+
+    proposer_index = get_beacon_proposer_index(state, state.slot)
+    if whistleblower_index is None:
+        whistleblower_index = proposer_index
+    whistleblowing_reward = slashed_balance // WHISTLEBLOWING_REWARD_QUOTIENT
```

Comment:
Not quite understand the renaming from `WHISTLEBLOWER_REWARD_QUOTIENT` to `WHISTLEBLOWING_REWARD_QUOTIENT` and `whistleblower_reward` to `whistleblowing_reward`; I thought the symmetry between `proposer` and `whistleblower` looks good? 😄

Reply:
Ah, second look, understand that you mean `whistleblowing_reward - proposer_reward` is the `whistleblower_reward` here.

Reply:
Right. There's a whistleblow**ing** reward that is split into two sub-rewards: a whistleblow**er** reward, and a proposer reward.